web-components/src/components/folksnomy/folksonomy-property-products.ts

```diff
@@ -0,0 +1,677 @@
+      console.error("Error fetching property products:", error)
+      this.error = "Failed to load products. Please try again later."
+    } finally {
+      this.loading = false
+    }
+  }
+
+  private applyFilters() {
+    const { barcode, value } = this.filters
+
+    // Filter products view
+    this.filteredProducts = this.products.filter((item) => {
+      const matchesBarcode = !barcode || item.product.toLowerCase().includes(barcode.toLowerCase())
+      const matchesValue = !value || item.v.toLowerCase().includes(value.toLowerCase())
+      return matchesBarcode && matchesValue
+    })
+
+    // Filter values view (only value filter applies)
+    this.filteredValues = this.groupedValues.filter((item) => {
+      const matchesValue = !value || item.value.toLowerCase().includes(value.toLowerCase())
+      return matchesValue
+    })
+  }
+
+  private handleFilterInput(field: "barcode" | "value", value: string) {
+    this.filters = {
+      ...this.filters,
+      [field]: value,
+    }
+
+    // Debounce filtering with 300ms delay for live filtering
+    if (this.filterTimeout) {
+      clearTimeout(this.filterTimeout)
+    }
+
+    this.filterTimeout = window.setTimeout(() => {
+      this.applyFilters()
+    }, 300)
+  }
+
+  private resetFilters() {
+    this.filters = {
+      barcode: "",
+      value: "",
+    }
+    this.filteredProducts = [...this.products]
+    this.filteredValues = [...this.groupedValues]
+  }
+
+  private switchViewMode(mode: "products" | "values") {
+    this.viewMode = mode
+  }
+
+  private downloadCSV() {
+    // Create CSV content based on current view mode
+    const isProductsView = this.viewMode === "products"
+    const headers = isProductsView
+      ? ["Product Barcode", "Corresponding Value"]
+      : ["Value", "Product Count"]
+
+    const rows = isProductsView
+      ? this.filteredProducts.map((item) => [item.product, item.v])
+      : this.filteredValues.map((item) => [item.value, item.count])
+
+    const csvContent = [headers.join(","), ...rows.map((row) => row.join(","))].join("\n")
+
+    // Create blob and download
+    const blob = new Blob([csvContent], { type: "text/csv;charset=utf-8;" })
+    const link = document.createElement("a")
+    const url = URL.createObjectURL(blob)
+
+    const filename = isProductsView
+      ? `folksonomy_property_${this.propertyName}_products.csv`
+      : `folksonomy_property_${this.propertyName}_values.csv`
+
+    link.setAttribute("href", url)
+    link.setAttribute("download", filename)
+    link.style.visibility = "hidden"
+
+    document.body.appendChild(link)
+    link.click()
+    document.body.removeChild(link)
+  }
+
+  private getProductUrl(productCode: string) {
+    return `/product/${productCode}`
+  }
+
+  private getPropertyValueUrl(value: string) {
+    return `/property/${this.propertyName}/value/${value}`
+  }
+
+  private getDocumentationUrl() {
+    return `https://wiki.openfoodfacts.org/Folksonomy/Property/${this.propertyName}`
+  }
```

Comment:
we have the same thing in the other component I think.

Could we put those methods in api/folksonomy.ts or some shared utils ? (in a next PR).